eZ/Publish/Core/Persistence/Legacy/Content/Location/Trash/Handler.php

```diff
@@ -169,28 +172,30 @@ public function recover($trashedId, $newParentId)
     }
 
     /**
-     * Returns an array of all trashed locations satisfying the $criterion (if provided),
-     * sorted with SortClause objects contained in $sort (if any).
-     * If no criterion is provided (null), no filter is applied.
-     *
-     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion $criterion
-     * @param int $offset Offset to start listing from, 0 by default
-     * @param int $limit Limit for the listing. Null by default (no limit)
-     * @param \eZ\Publish\API\Repository\Values\Content\Query\SortClause[] $sort
-     *
-     * @return \eZ\Publish\SPI\Persistence\Content\Location\Trashed[]
+     * {@inheritdoc}.
      */
     public function findTrashItems(Criterion $criterion = null, $offset = 0, $limit = null, array $sort = null)
     {
+        // @TODO: This only works for direct SPI usage, any API/UI usage needs criteria to be taken into account so we
+        //        respect user rights here.
+        $totalCount = $this->locationGateway->countTrashed();
+        if (!$totalCount) {
```

Comment:
As `countTrashed` will always return `int` than maybe more clear will be `if ($totalCount === 0) {`